### ISSUE TITLE
Custom Value Transformation for Normalizers

### ISSUE THREAD

**Issue by Nephdz** [jdb78/pytorch-forecasting#1312]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0.1
- Python version: 3.10.11
- Operating System: Ubuntu 20.04.6 LTS

What's the difference between forward,reverse,inverse,inverse_troch for the transformations? I'm trying to make a custom sigmoid/softmax transformation and I don't understand why there are all these options. In the code, in seems like "forward" and "reverse" functions are applied to the values and "inverse_torch" is used for metrics. It says that "inverse" is used if "forward" isn't the inverse of "reverse" but I don't see any references to "inverse" being called.

https://github.com/jdb78/pytorch-forecasting/blob/7c775c1a2fe3effe62eac0a476db98ca5b24ca4c/pytorch_forecasting/data/encoders.py#LL130C1-L146C6
```
class TransformMixIn:
    """Mixin for providing pre- and post-processing capabilities to encoders.

    Class should have a ``transformation`` attribute to indicate how to preprocess data.
    """

    # dict of PyTorch functions that transforms and inversely transforms values.
    # inverse entry required if "reverse" is not the "inverse" of "forward".
    TRANSFORMATIONS = {
        "log": dict(forward=_clipped_log, reverse=torch.exp, inverse_torch=ExpTransform()),
        "log1p": dict(forward=torch.log1p, reverse=torch.exp, inverse=torch.expm1, inverse_torch=Expm1Transform()),
        "logit": dict(forward=_clipped_logit, reverse=_clipped_sigmoid, inverse_torch=SigmoidTransform()),
        "count": dict(forward=_plus_one, reverse=F.softplus, inverse=_minus_one, inverse_torch=MinusOneTransform()),
        "softplus": dict(forward=softplus_inv, reverse=F.softplus, inverse_torch=SoftplusTransform()),
        "relu": dict(forward=_identity, reverse=F.relu, inverse=_identity, inverse_torch=ReLuTransform()),
        "sqrt": dict(forward=torch.sqrt, reverse=_square, inverse_torch=PowerTransform(exponent=2.0)),
    }
```

A strange case in the code would also be specifying the "logit" transformation. It has a "forward" of torch.log1p and I would expect it to have torch.expm1 as it's "reverse" but it's instead set to torch.exp. "Inverse" and "inverse_torch" are both expm1.
